lib/plugins/aws/package/compile/events/stream.js

```diff
@@ -16,72 +16,83 @@ class AwsCompileStreamEvents {
-            bisectBatchOnFunctionError: { type: 'boolean' },
-            maximumRecordAgeInSeconds: {
-              anyOf: [{ const: -1 }, { type: 'integer', minimum: 60, maximum: 604800 }],
-            },
-            functionResponseType: { enum: ['ReportBatchItemFailures'] },
-            destinations: {
+          allOf: [
+            {
               type: 'object',
               properties: {
-                onFailure: {
+                arn: {
                   anyOf: [
+                    { $ref: '#/definitions/awsCfInstruction' },
                     { $ref: '#/definitions/awsArnString' },
-                    {
-                      type: 'object',
-                      properties: {
-                        // arn constraints are listed in oneOf property of this schema
-                        arn: {},
-                        type: { enum: ['sns', 'sqs'] },
-                      },
-                      additionalProperties: false,
+                  ],
+                },
+                type: { enum: ['dynamodb', 'kinesis'] },
+                batchSize: { type: 'integer', minimum: 1, maximum: 10000 },
+                parallelizationFactor: { type: 'integer', minimum: 1, maximum: 10 },
+                startingPosition: { enum: ['LATEST', 'TRIM_HORIZON'] },
+                enabled: { type: 'boolean' },
+                consumer: { anyOf: [{ const: true }, { $ref: '#/definitions/awsArn' }] },
+                batchWindow: { type: 'integer', minimum: 0, maximum: 300 },
+                maximumRetryAttempts: { type: 'integer', minimum: -1, maximum: 10000 },
+                bisectBatchOnFunctionError: { type: 'boolean' },
+                maximumRecordAgeInSeconds: {
+                  anyOf: [{ const: -1 }, { type: 'integer', minimum: 60, maximum: 604800 }],
+                },
+                functionResponseType: { enum: ['ReportBatchItemFailures'] },
+                destinations: {
+                  type: 'object',
+                  properties: {
+                    onFailure: {
                       anyOf: [
+                        { $ref: '#/definitions/awsArnString' },
                         {
+                          type: 'object',
                           properties: {
-                            arn: { $ref: '#/definitions/awsCfFunction' },
+                            // arn constraints are listed in oneOf property of this schema
+                            arn: {},
+                            type: { enum: ['sns', 'sqs'] },
                           },
-                          required: ['arn', 'type'],
-                        },
-                        {
-                          properties: {
-                            arn: { $ref: '#/definitions/awsArnString' },
-                          },
-                          required: ['arn'],
+                          additionalProperties: false,
+                          anyOf: [
```

Comment:
I'm a little bit confused about this change/improvement. Why do we now have to list this `anyOf` twice in schema? I would also appreciate a high-level reasoning on how it improves the schema because honestly by just looking at it, it's more confusing than before

Reply:
Sorry - it should be just a comment, without "request changes" :grimacing: